hyp3_gamma/__main__.py

```diff
@@ -33,10 +34,20 @@ def main():
     )
 
 
+def write_earthdata_creds(username, password):
+    if username is None:
+        username = os.environ['EARTHDATA_USERNAME']
```

Comment:
I believe this will throw a `KeyError` if the environment variable is not defined? Should we give the user an explicit error message, i.e. "earthdata login username must be provided via either the --username option or an EARTHDATA_USERNAME environment variable"?

Reply:
Good point. Would it be best to write that message to stderr?

Reply:
We typically `raise` a python exception with a custom error message, e.g. 
https://github.com/ASFHyP3/hyp3-gamma/blob/develop/hyp3_gamma/rtc/rtc_sentinel.py#L91
https://github.com/ASFHyP3/hyp3-gamma/blob/develop/hyp3_gamma/rtc/rtc_sentinel.py#L138
https://github.com/ASFHyP3/hyp3-gamma/blob/develop/hyp3_gamma/rtc/rtc_sentinel.py#L180